src/interceptors/fetch/index.ts

```diff
@@ -40,7 +40,13 @@ export class FetchInterceptor extends Interceptor<HttpRequestEventMap> {
     globalThis.fetch = async (input, init) => {
       const request = new Request(input, init)
 
-      const url = typeof input === 'string' ? input : input.url
+      const url =
+        typeof input === 'string'
+          ? input
+          : input instanceof URL
+          ? input.href
+          : input.url
+
```

Comment:
I believe this is the one user-facing change, to select `.href` in the case that `input` is an instance of `URL`.

I don't fully grok the root cause of this - update to JSDom maybe? But I note that it fixed a type error, which was reported in my VSCode with latest TypeScript but not by any build script. It also fixed some tests, but I can't remember exactly which, since this was one of multiple bugs fixed simultaneously.

It looks like a safe change to me, though, since the new branch should only evaluate when `input instanceof URL`, which previously would only result in an error since [there is no valid `url` property on an instance of `URL`](https://developer.mozilla.org/en-US/docs/Web/API/URL). And it still works on Node v14/v16, although that's not unexpected if the root cause was an update to `jest-environment-jsdom`.

Reply:
Do we need this change for type compliance only? Afaik Fetch supports `URL` as input. Is that no longer the case for Node.js fetch?

Reply:
Good question. I'm looking more closely at this now. Reverting this change does not cause any tests to fail on my local (non-IPv6) system. I also checked in GitHub Actions, and no tests failed there either. So we could revert it, but let me first check if it breaks any tests in the next branch following this one. I *think* it could fix a real issue that was only *exposed* by a TypeScript error, but I don't have any regression test to prove that.

The reason it's a type error in VSCode but not any of the build scripts is that VSCode uses version 4.7.3 while this package uses version 4.3.5, and the type error is related to the latest version of the `fetch` API from `lib.dom.d.ts`. So the type error is likely indicating a real bug, even if there are no tests for it. Likely the bug would surface in some environments (browser vs. Node) but not others.

Specifically, the error is:

```
Property 'url' does not exist on type 'Request | URL'.
  Property 'url' does not exist on type 'URL'.ts(2339)
```

as reported in VSCode:

![image](https://user-images.githubusercontent.com/835921/193116143-995b014c-9823-4d20-bb8b-1c2267777cdb.png)

Note that it's inferring the type of `input` as `RequestInfo` which it infers from the definition of `globalThis.fetch` in the latest TypeScript `lib.dom.d.ts`:

```ts
declare function fetch(input: RequestInfo | URL, init?: RequestInit): Promise<Response>;
```

visually:

![image](https://user-images.githubusercontent.com/835921/193116548-3048fe8a-f4a2-4679-bd8a-e6417d442b44.png)

Whether or not this indicates a bug at runtime depends on the `fetch` API provided by the environment. If we're using a subset of that API, then we should use module augmentation to keep the TypeScript API consistent with what we expect at runtime (although admittedly it won't be an issue until we upgrade the version of TypeScript used in the build scripts).

Reply:
Ok, so reverting this change does have an effect on a downstream dependency - namely, the msw branch where I'm adding support for node-native interception. It causes [tests to fail on IPv6 systems](https://github.com/milesforks/msw/actions/runs/3154340675/jobs/5131825375#step:7:16) that were otherwise already fixed with [this commit](https://github.com/milesforks/msw/pull/2/commits/4ded61dcb25ccac51c86d4bd52275697bf9e408d).

So, from a backwards compatibility perspective, it would be safer to revert this change. But from a forward compatibility perspective, we need it if we want to operate properly on RequestInfo objects in Node 18. The balance seems to weigh in favor of keeping the commit.

If you're okay with keeping this, then the PR is ready to merge